### PR TITLE
Add a connection pool metric for connections in establishment.

### PR DESCRIPTION
**Pull request by mikkokar**
Exposes a number of connections being under established, per each connection pool.

### CODE REVIEW

File: components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java [hunk 1]
```diff
@@ -81,11 +81,16 @@ public void close() {
...
+    private static void registerMetrics(ConnectionPool hostConnectionPool, MetricRegistry scopedRegistry) {
+        ConnectionPool.Stats stats = hostConnectionPool.stats();
+        scopedRegistry.register("busy-connections", (Gauge<Integer>) stats::busyConnectionCount);
+        scopedRegistry.register("pending-connections", (Gauge<Integer>) stats::pendingConnectionCount);
+        scopedRegistry.register("available-connections", (Gauge<Integer>) stats::availableConnectionCount);
+        scopedRegistry.register("connection-attempts", (Gauge<Integer>) () -> (int) stats.connectionAttempts());
+        scopedRegistry.register("connection-failures", (Gauge<Integer>) () -> (int) stats.connectionFailures());
+        scopedRegistry.register("connections-closed", (Gauge<Integer>) () -> (int) stats.closedConnections());
+        scopedRegistry.register("connections-terminated", (Gauge<Integer>) () -> (int) stats.terminatedConnections());
+        scopedRegistry.register("in-establishment", (Gauge<Integer>) () -> (int) stats.connectionsInEstablishment());
```
Comment (dvlato) [inline components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java]: Should we keep the same terminology here? All the others have 'connections' in the name.

File: components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java [hunk 2]
```diff
@@ -98,15 +103,12 @@ private void registerMetrics() {
...
-    private static void registerMetrics(ConnectionPool hostConnectionPool, MetricRegistry scopedRegistry) {
-        ConnectionPool.Stats stats = hostConnectionPool.stats();
-        scopedRegistry.register("busy-connections", (Gauge<Integer>) stats::busyConnectionCount);
-        scopedRegistry.register("pending-connections", (Gauge<Integer>) stats::pendingConnectionCount);
-        scopedRegistry.register("available-connections", (Gauge<Integer>) stats::availableConnectionCount);
-        scopedRegistry.register("connection-attempts", (Gauge<Integer>) () -> (int) stats.connectionAttempts());
-        scopedRegistry.register("connection-failures", (Gauge<Integer>) () -> (int) stats.connectionFailures());
-        scopedRegistry.register("connections-closed", (Gauge<Integer>) () -> (int) stats.closedConnections());
-        scopedRegistry.register("connections-terminated", (Gauge<Integer>) () -> (int) stats.terminatedConnections());
+    private void removeMetrics() {
```
Comment (dvlato) [inline components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java]: I would be happier if we didn't have to repeat the list of metrics here.  Accepting for now...

Comment (mikkokar) [reply components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java]: I agree.

File: components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java (hunk 1 above)
Comment (mikkokar) [reply components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java]: I will change this later when fixing the pool error.

**Pull request merged**